src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -538,6 +545,38 @@ def compile(
             },
         )()
 
+    def always_do_after_compile(self, ctx):
+        self._pass_diagnostics_to_buildstats(ctx)
+
+    def _aggregate_diagnostics(self, lsp_data):
+        # Note: this is not arbitrary. It is exactly every value that the LSP's DiagnosticSeverity allows.
+        # See https://microsoft.github.io/language-server-protocol/specification#diagnostic
+        counts = {
```

Comment:
Where would warnings-treated-as-errors end up?

Reply:
I think they would end up as errors. TBH, I think Errors are handled with some exceptions, so I'm not even sure the errors count will be working properly; but this is still the correct thing to do in theory as this is parsing LSP compliant data, which is what we output. Let me know if this doesn't make sense.